test/swar/BasicOperations.cpp

```diff
@@ -323,3 +337,76 @@ constexpr auto aBooleansWithTrue = booleans(SWAR<4, u32>{0x1});
 static_assert(aBooleansWithTrue);
 static_assert(!aBooleansWithTrue); // this is a pitfall, but lesser evil?
 static_assert(false == !bool(aBooleansWithTrue));
+
+TEST_CASE(
+    "fullAddition",
+    "[swar]"
+) {
+    {
+        // 0x0111 (7) + 0x0001 (1) is 0x1000 (-8)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'1000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.result.value());
+    }
+    {
```

Comment:
These tests surface the problem that we don't have an ergonomic way to put negative numbers in lanes, in this example, neither `BitmaskMaker` nor `broadcast` will do the right thing if the argument is -8.